app/src/main/java/com/jerboa/MainActivity.kt

```diff
@@ -115,10 +108,17 @@ class MainActivity : AppCompatActivity() {
                 null
             }
 
-            LaunchedEffect(Unit) {
-                fetchInitialData(accountSync, siteViewModel)
+            val accountViewModel: AccountViewModel = viewModel(factory = AccountViewModelFactory.Factory)
+
+            val account = getCurrentAccount(accountViewModel)
+
+            LaunchedEffect(account) {
+                if (account == null || account.id != -1) {
```

Comment:
There was an occasional race condition type issue where `getCurrentAccount` returns an initial value of `null` because it hasn't gotten the data from the database yet. Since there was no initial state set on the `observeAsState` method here https://github.com/dessalines/jerboa/pull/973/files#diff-697367ddff23d571ded643b597f312aeba4b29669b058e9edb03edf9bb030d92R14, it was making two calls to `fetchInitialData`. Once with the `null` initial value, then again with the actual data from the database. This caused two site data calls to go out, depending on which one returned first, the user data in the siteViewModel could set incorrectly set to the wrong value.

As a solution to this, I set an initial state value in  `observeAsState`, so I know if the account id is -1, we haven't actually gotten any data yet from Room, so there is no need to call `fetchInitialData`. If there is no account in Room, it will return `null` and work like before. 

Let me know if there's a better way to handle something like this

Reply:
Seems fine to me.